bundles/http/Citrix/Citrix-Virtual-Apps-and-Desktops-service/7687284a-09a1-4494-b33c-f6bfa0932656.js

```diff
@@ -245,6 +270,121 @@ function flatJson(json, oName) {
     return json
 }
 
+// Actions functions
+
+async function toggleMaintenanceMode(params) {
+    const {actionParameters, integrationParameters, client} = params
+    const path = `${actionParameters.siteId}/Machines/${actionParameters.id}`
+    const body = {
+        InMaintenanceMode: actionParameters.inMaintenanceMode,
+    }
+
+    const response = await client.fetch(path, {
+        method: 'PATCH',
+        body: JSON.stringify(body),
+        headers: {
+            'Citrix-CustomerId': integrationParameters.customerId,
+        },
+    })
+
+    if (!response.ok) {
+        throw new Error(
+            `Setting maintenance mode failed (${response.status}: ${response.statusText})`,
+        )
+    }
+}
+
+async function rebootMachine(params) {
+    const {actionParameters, integrationParameters, client} = params
+    const path = `${actionParameters.siteId}/Machines/${actionParameters.id}/$reboot?force=${actionParameters.force}`
+
+    const response = await client.fetch(path, {
+        method: 'POST',
+        headers: {
+            'Citrix-CustomerId': integrationParameters.customerId,
+        },
+    })
+
+    if (!response.ok) {
+        throw new Error(`Machine reboot failed (${response.status}: ${response.statusText})`)
+    }
+}
+
+async function updateMachine(params) {
+    const {actionParameters, client, integrationParameters, dataStore} = params
+    const path = `${actionParameters.siteId}/Machines/${actionParameters.id}`
+
+    let machine = await client.fetch(path, {
```

Comment:
Haven't you forgotten
```js
    method: 'POST',
```
here?

Reply:
It's GET request, i'm updating the machine in our cache, not it CVAD.